Integrate feedback from the release of 0.1.0 (#194)
wireapp/wire-docs: src/understand/federation/architecture.rst

Backend domains
^^^^^^^^^^^^^^^

Each backend has two domains: an `infrastructure domain` and a `backend domain`.

The former is the domain under which the backend is actually reachable via the

> network. It is also the domain as which each backend authenticates itself
> towards other backends.

Reviewer: ```suggestion
network. It is also the domain that each backend uses in authenticating itself
to other backends.
```

This step is necessary in two scenarios:

* A backend would like to establish a connection to another backend that it only
  knows the backend domain of. This is the case, for example, when a user of a
  local backend searches for a :ref:`qualified username <qualified-user-name>`,

>   which only includes that user's backend's backend domain.

Reviewer: Doesn't "backend's backend domain" sound strange? Perhaps we could rename "backend" in "backend domain" to something else. Maybe "external domain", and then the infra domain would be "internal domain"?

Author: Yes, renaming "backend domain" would be sensible. What do you mean by "external" in "external domain"? I would have gone for something like "pretty domain" or "presentation domain" or something similar indicating its purpose in the context of federation.

Reviewer: I agree "external" is not the best. What I meant is that it is a publicly-facing domain, which is from the backend's point of view for communication with the external world.

Author: I will come up with something and propose a change, but not in this PR.
